### PR TITLE
Add a project setting to configure the maximum number of timestamps.

### LINKED ISSUE #91531: Running more than one viewport triggers an error in the Visual Debugger because you run out of max_timestamp_query_elements
### Tested versions

Reproducible in:

v4.3.dev.custom_build [https://github.com/godotengine/godot/commit/479b2ab777188e21109bcfe803c2599c35bb9042]
v4.3.dev6.official [https://github.com/godotengine/godot/commit/89850d553eeb259e208d0c577cd7bc1eabd3a90a]

### System information

Godot v4.3.dev (479b2ab) - Windows 10.0.22631 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 4090 (NVIDIA; 31.0.15.2824) - Intel(R) Core(TM) i9-14900KF (32 Threads)  Issue description

### Issue description

When I run my game with more than one viewport active (a typical setup for a 3D racing game where another viewport is used to render mirrors and optionally two more for triple screen setups with 3 viewports instead of 1) and activate the Visual Debugger, I run into the issue that by default there are not enough timestamp query elements available (see screenshot). I've talked to @DarioSamo about this while discussing a fix for https://github.com/godotengine/godot/issues/91517 and the solution was to increase the limit from 256 (in rendering_device.cpp:4967 and in utilities.h:183) to 2048. He was not sure why the initial limit of 256 was chosen, so that is one question that needs to be answered. Another option we discussed in RocketChat was to make this setting configurable as a project setting (possibly requiring a restart of the editor). Open to other suggestions too obviously!

capture_timestamp_2

### Steps to reproduce

I did not make a MRP as @DarioSamo claimed it would probably be overkill right now.

### Minimal reproduction project (MRP)

(see steps to reproduce above)

### PR DESCRIPTION
**Pull request by m4rr5**
With a description and a reference to the setting when the limit is hit and an ERR_FAIL_COND is hit.

This is a fix for: https://github.com/godotengine/godot/issues/91531

* *Bugsquad edit, fixes: https://github.com/godotengine/godot/issues/91531*

### CODE REVIEW

File: doc/classes/ProjectSettings.xml [hunk 1]
```diff
@@ -620,6 +620,9 @@
 		<member name="debug/settings/profiler/max_functions" type="int" setter="" getter="" default="16384">
 			Maximum number of functions per frame allowed when profiling.
 		</member>
+		<member name="debug/settings/profiler/max_timestamp_query_elements" type="int" setter="" getter="" default="256">
+			Maximum number of timestamp query elements per frame allowed when visual profiling.
```
Comment (AThousandShips) [inline doc/classes/ProjectSettings.xml:624]:
```suggestion
			Maximum number of timestamp query elements allowed per frame for visual profiling.
```
Less dense sentence, might need to be improved further

File: servers/rendering/rendering_device.cpp [hunk 2]
```diff
@@ -5208,7 +5208,7 @@ void RenderingDevice::_free_rids(T &p_owner, const char *p_type) {
 void RenderingDevice::capture_timestamp(const String &p_name) {
 	ERR_FAIL_COND_MSG(draw_list != nullptr, "Capturing timestamps during draw list creation is not allowed. Offending timestamp was: " + p_name);
 	ERR_FAIL_COND_MSG(compute_list != nullptr, "Capturing timestamps during compute list creation is not allowed. Offending timestamp was: " + p_name);
-	ERR_FAIL_COND(frames[frame].timestamp_count >= max_timestamp_query_elements);
+    ERR_FAIL_COND_MSG(frames[frame].timestamp_count >= max_timestamp_query_elements, vformat("Tried capturing more timestamps than the configured maximum (%d). You can increase this limit in the project settings under 'Debug/Settings' called 'Max Timestamp Query Elements'.", max_timestamp_query_elements));
```
Comment (AThousandShips) [inline servers/rendering/rendering_device.cpp:5211]:
```suggestion
	ERR_FAIL_COND_MSG(frames[frame].timestamp_count >= max_timestamp_query_elements, vformat("Tried capturing more timestamps than the configured maximum (%d). You can increase this limit in the project settings under 'Debug/Settings' called 'Max Timestamp Query Elements'.", max_timestamp_query_elements));
```

**Review by Mickeon — APPROVED** [review]
Documentation is fine.

It seems like it was @clayjohn 's idea to [make it a project setting](https://github.com/godotengine/godot/issues/91531#issuecomment-2093675452). I wonder what he thinks of it now.

As for me, I heavily appreciate the more explicit error message, although I really worry it will be spammed a lot.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**